### PR TITLE
[Community][OAUTH2] Fix auth2 login test cases

### PR DESCRIPTION
**Pull request by afabiani**
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java [hunk 1]
```diff
@@ -13,10 +13,7 @@
  */
 package org.geoserver.security.oauth2;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertSame;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (aaime) [inline src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java:16]: Don't use star imports (GeoServer general policy, normally enforced by QA)

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java [hunk 2]
```diff
@@ -98,13 +95,12 @@ public void check(OpenIdConnectFilterConfig config) throws Exception {
...
 
         config.setRoleConverterName(null);
-
         config.setCheckTokenEndpointUrl(null);
 
         failed = false;
         try {
             validator.validateOAuth2FilterConfig(config);
-        } catch (OAuth2FilterConfigException ex) {
+        } catch (FilterConfigException ex) {
```
Comment (aaime) [inline src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java:103]: Is this the actual fix?

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-web/src/test/java/org/geoserver/web/security/oauth2/OpenIdConnectLoginButtonTest.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public void testLoginButton() {
         // the login form is there and has the link
         assertTrue(
                 html.contains(
-                        "<form style=\"display: inline-block;\" method=\"post\" action=\"http://localhost/context/web/j_spring_oauth2_openid_connect_login\">"));
```
Comment (aaime) [inline src/community/security/oauth2-openid-connect/oauth2-openid-connect-web/src/test/java/org/geoserver/web/security/oauth2/OpenIdConnectLoginButtonTest.java:60]: Good, we need to get rid of local styles anyways for Wicket 9 upgrade.

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java (hunk 1 above)
Comment (afabiani) [reply src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java]: Reverted

File: src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java (hunk 2 above)
Comment (afabiani) [reply src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/test/java/org/geoserver/security/oauth2/OAuth2FilterConfigValidatorTest.java]: Reverted

**Comment by aaime** [comment]
Looks good to me, let's check the results of the github checks

**Pull request merged**